### PR TITLE
LPS-164598: Test Fix com.liferay.headless.admin.user.resource module

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Changes in `OrganizationResourceTest.testDeleteAccountByExternalReferenceCodeOrganization` and `OrganizationResourceTest.testDeleteAccountOrganization` tests cases
- Changes in `AccountRoleResourceTest.testDeleteAccountByExternalReferenceCodeAccountRoleUserAccountByEmailAddress` test case
- Each of them returned a status code 404 but now is a 204 one.

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-164598

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java [hunk 1]
```diff
@@ -268,18 +268,22 @@ protected String[] getAdditionalAssertFieldNames() {
 			testDeleteAccountByExternalReferenceCodeOrganization_addOrganization()
 		throws Exception {
 
-		return organizationResource.putOrganizationByExternalReferenceCode(
-			StringUtil.toLowerCase(RandomTestUtil.randomString()),
-			randomOrganization());
+		Organization organization =
+			organizationResource.putOrganizationByExternalReferenceCode(
+				_accountEntry.getExternalReferenceCode(), randomOrganization());
```
Comment (ccorreagg) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java:273]: Why are you using the ERC of the account entry to assign it to the organization?

**Review by ccorreagg — CHANGES_REQUESTED** [review]
I think we should not assign the same ERC, please review it

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/OrganizationResourceTest.java]: My fault, I saw it in other tests but it make sense, I'm going to correct it

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by ccorreagg — COMMENTED** [review]
LGTM!

Should we include the team responsible of user administration in CC?

**Review by ccorreagg — APPROVED** [review]

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**